OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -151,6 +161,17 @@ public Cargo(ActorInitializer init, CargoInfo info)
 				totalWeight = cargo.Sum(c => GetWeight(c));
 			}
 
+			foreach (var c in cargo)
+			{
+				c.Trait<Passenger>().Transport = self;
+
+				foreach (var nec in c.TraitsImplementing<INotifyEnteredCargo>())
```

Comment:
Are we sure it is a good idea to make these calls before `Created` ran? (E.g. this might/wll break with https://github.com/OpenRA/OpenRA/blob/ae34410c8077a45d3dae20419aa63659cf0718a6/OpenRA.Mods.Common/Traits/Passenger.cs#L75-L78 and https://github.com/OpenRA/OpenRA/blob/ae34410c8077a45d3dae20419aa63659cf0718a6/OpenRA.Mods.Common/Traits/Passenger.cs#L123-L126 won't it?)

Reply:
Dont think so, INotifyCreated.Created will be called in the call chain `World.CreateActor() -> Actor.Created()` Right?
https://github.com/OpenRA/OpenRA/blob/1027b125e13e8efa47dddd7492dad71d8683cbf0/OpenRA.Mods.Common/Traits/Cargo.cs#L143-L144
or
https://github.com/OpenRA/OpenRA/blob/1027b125e13e8efa47dddd7492dad71d8683cbf0/OpenRA.Mods.Common/Traits/Cargo.cs#L155-L156

Reply:
Ah yes, you're right, they probably run for each actor and not once all actors are done.